qiskit/circuit/commutation_checker.py

```diff
@@ -20,17 +20,31 @@
 from qiskit.circuit import Qubit
 from qiskit.circuit.operation import Operation
 from qiskit.circuit.controlflow import CONTROL_FLOW_OP_NAMES
+from qiskit.circuit.library import XGate, YGate, ZGate
+from qiskit.circuit.library.generalized_gates.pauli import PauliGate
 from qiskit.quantum_info.operators import Operator
 
 _skipped_op_names = {"measure", "reset", "delay", "initialize"}
 _no_cache_op_names = {"annotated"}
 
+_rotation_gates = {
+    "rx": XGate(),
```

Comment:
Could we add controlled Pauli rotations and the phase gate (-> RZ) as well?

Reply:
Yes, this should be trivial. I can look into all the paramterized gates. As long as they span from a single gate rotation it is as easy as making this dictionary bigger.